src/routes/transactions.js

```diff
@@ -134,4 +134,37 @@ server.post("/transaction", isAuth, async (req, res) => {
   }
 });
 
+server.get('/transaction-history', isAuth, async (req, res) => {
+  const userId = req.session.passport.user._id;
+  const transactions = await Transaction.find({userId: userId});
+  const response = [];
+
+  if(transactions) {
+    for (const transaction of transactions) {
+      const baseCurrency = await Currency.findById(transaction.baseCurrencyId);
+      const exchangeCurrency = await Currency.findById(transaction.exchangeCurrencyId);
+
+      response.push({
+        baseCurrencyName: baseCurrency.currencyName,
+        exchangeCurrencyName: exchangeCurrency.currencyName,
+        baseCurrencyAmount: transaction.baseCurrencyAmount,
+        exchangeCurrencyAmount: transaction.exchangeCurrencyAmount,
+        availableExchangeAmount: transaction.availableExchangeAmount,
+        cryptoInWallet: transaction.cryptoInWallet,
+        currencyInWallet: transaction.currencyInWallet,
+        transactionDate: transaction.transactionDate
+      })
+    }
+
+    res.status(200).json({
```

Comment:
To be consistent with returns in the function you cand use return here

```suggestion
    return res.status(200).json({
```